Source/SIMPLib/Geometry/IGeometryGrid.h

```diff
@@ -97,6 +97,11 @@ class SIMPLib_EXPORT IGeometryGrid : public IGeometry
   virtual void getCoords(size_t x, size_t y, size_t z, double coords[3]) const = 0;
   virtual void getCoords(size_t idx, double coords[3]) const = 0;
 
+  virtual std::optional<size_t> getIndex(float coords[3]) const = 0;
+  virtual std::optional<size_t> getIndex(float xCoord, float yCoord, float zCoord) const = 0;
+  virtual std::optional<size_t> getIndex(double coords[3]) const = 0;
+  virtual std::optional<size_t> getIndex(double xCoord, double yCoord, double zCoord) const = 0;
```

Comment:
```
virtual std::optional<size_t> getIndex(float coords[3]) const = 0;
virtual std::optional<size_t> getIndex(double coords[3]) const = 0
```

I know this is the same kind of signature as getCoords() has above, but do we need it? If so, it should probably be a std::array or a span (if we add that to the SDK) depending on the exact use case.

Reply:
Or can we just use a float* with the documentation saying the pointer is a 3 element array? Sometimes we have millions/billions of coords during a runtime and I would rather not incur the penalty of dynamic casting? Thoughts.

Reply:
This is where a span would be the best option especially since we're trying to move away from raw pointers.
`std::span<const float, 3>` is just a pointer and a length under the hood just with the size enforced at compile time.

Also `std::optional<size_t> getIndex(float coords[3]) const` just calls `std::optional<size_t> getIndex(float xCoord, float yCoord, float zCoord) const` anyways in the current implementation.

Reply:
`std::optional<size_t> getIndex(float coords[3]) const` is treated as `std::optional<size_t> getIndex(float* coords) const` by the compiler but is more descriptive to the user about what is expected. Making the code less descriptive and more reliant on comments is going the wrong way.

I don't feel strongly about span, but it does seem to be the next step in improving upon how we pass data.

Reply:
I just want to make sure that whatever we chose is the "right" way update the API. My guess is that if we decide to update the API these are non-trivial changes to the dependent codes. I could be wrong. there may not be a lot of filters that depend on these APIs.

Reply:
I'm for having expressive APIs but not at the detriment to the developer understanding of the API itself.

Reply:
So which way are we going with this.  We need to make a decision so that we can merge it in.

Reply:
We don't necessarily have to update the old API, and we can just add the new getIndex() methods. We could discuss at the meeting as well.

Reply:
We discussed at the meeting and decided to remove the array method declarations for now.